lib/git-process.ts

```diff
@@ -311,3 +332,33 @@ function ignoreClosedInputStream(process: ChildProcess) {
     }
   })
 }
+
+export interface IGitTask {
+  pid?: number
+  readonly result: Promise<IGitResult>
+  setPid(pid: number): void
+  readonly cancel: () => boolean
+}
+
+class GitTask implements IGitTask {
+  constructor(result: Promise<IGitResult>) {
+    this.result = result
+  }
+
+  pid?: number
+  result: Promise<IGitResult>
+  public setPid(pid: number) {
+    this.pid = pid
+  }
+  public cancel(): boolean {
+    if (this.pid !== undefined) {
+      try {
+        fs.process.kill(this.pid)
+        return true
+      } catch (e) {
+        return false
+      }
+    }
+    return false
```

Comment:
@maifeeulasad in your opinion, why do we need to return true/false? Why do you think is useful? I'm just curious about what kind of usage are you anticipating!